File: Jogging/jogging.ts
const func = (seconds: number, speed: number, rest: number, X: number) => {
  let time = 0
  let distance = 0

  while (true) {
    for (let i = 0; i < seconds; i++) {
      distance += speed
      time++
      if (time === X) {
        return distance
      }
    }

    for (let i = 0; i < rest; i++) {
      time++
      if (time === X) {
        return distance
      }
    }
  }
}

export const jogging = (input: string) => {
  const [tSeconds, tSpeed, tRest, aSeconds, aSpeed, aRest, X] = input.split(" ").map(Number)

  const takahashi = func(tSeconds, tSpeed, tRest, X)
  const aoki = func(aSeconds, aSpeed, aRest, X)

  if (takahashi > aoki) {
    return "Takahashi"
  } else if (takahashi < aoki) {
    return "Aoki"
  } else {
    return "Draw"
  }
}

File: Jogging/jogging.py
def jog(seconds, speed, rest, X):
    time = 0
    distance = 0

    while True:
        for i in range(seconds):
            distance += speed
            time += 1
            if time == X:
                return distance
        
        for i in range(rest):
            time += 1
            if time == X:
                return distance

def jogging(input):
    [tSeconds, tSpeed, tRest, aSeconds, aSpeed, aRest, X] = [int(s) for s in input.split(" ")]

    takahashi = jog(tSeconds, tSpeed, tRest, X)
    aoki = jog(aSeconds, aSpeed, aRest, X)

    if takahashi > aoki:
        return "Takahashi"
    elif takahashi < aoki:
        return "Aoki"
    else:
        return "Draw"
